src/daemon/ca/rc.rs

```diff
@@ -301,40 +301,20 @@ impl ResourceClass {
                 rcvd_cert.validity().not_after().to_rfc3339()
             );
 
+            // Prep certified key for updated received certificate
+            let certified_key = CertifiedKey::create(rcvd_cert);
+
             // Check whether child certificates should be shrunk
-            //
-            // NOTE: We need to pro-actively shrink child certificates to avoid invalidating them.
-            //       But, if we gain additional resources it is up to child to request a new certificate
-            //       with those resources.
-            //
-            let mut updates = ChildCertificateUpdates::default();
-            for issued in self.certificates.overclaiming(rcvd_resources) {
-                let remaining_resources = issued.resource_set().intersection(rcvd_resources);
-                if remaining_resources.is_empty() {
-                    // revoke
-                    updates.remove(issued.subject_key_identifier());
-                } else {
-                    // re-issue
-                    let re_issued = self.re_issue(
-                        issued,
-                        Some(remaining_resources),
-                        current_key,
-                        None,
-                        &config.issuance_timing,
-                        signer,
-                    )?;
-                    updates.issue(re_issued);
-                }
-            }
+            let updates = self
+                .certificates
+                .shrink_overclaiming(&certified_key, &config.issuance_timing, signer)?;
```

Comment:
The old code "shrunk" based on `current_key` but the new code moves creation of `certified_key` up and shrinks based on that, is this difference correct?

Reply:
logic was correct, but the naming was confusing in this context - hopefully better now